Release v0.6

jongpie opened the pull request.

jongpie committed:
Added first attempt at setting up .travis.yml

jongpie committed:
Added some echo statements to check some variable values

jongpie committed:
Removed the gender tag in NebulaLog__c.object because it's the worst/breaks deploys

jongpie committed:
Fixed a problem with SObjectRecordTypes when the object has no record types

jongpie committed:
Remove the echo statements I added to help debug - deployments are started to work now

jongpie committed:
Added Travis CI build badges to README.md

jongpie committed:
Added .travis.yml to the repo & configured Travis CI

jongpie committed:
Added QueryOperator_Tests.cls

jongpie committed:
Added license info to QueryOperator_Tests

jongpie committed:
Created NebulaSettings_Tests.cls

jongpie committed:
Fixed the ordering of the filter scopes to be in descending order

jongpie committed:
Changed NebulaSettings.cls to be static variables and methods

jongpie committed:
Added more tests to SObjectRepository_Tests.cls

jongpie committed:
Renamed the methods in Environment_Tests.cls

jongpie committed:
Added QueryUtils_Tests.cls

jongpie committed:
Fixed naming convention on a unit test

jongpie committed:
Added asserts and new methods to SObjectRepository_Tests.cls

jongpie committed:
Removed some extra line breaks in QueryUtils_Tests.cls

jongpie committed:
Added more unit tests & test classes

jongpie committed:
Changed Logger.cls to use static methods - everything now calls Logger directly

jongpie committed:
Logger now checks the custom setting to see if it is enabled

jongpie committed:
Renamed QueryUtils to QueryArgumentFormatter.cls

jongpie committed:
QueryArgumentFormatter no longer uses static methods

jongpie committed:
Removed an extra linebreak in NebulaSettings_Tests.cls

jongpie committed:
QueryBuilder.cls now uses QueryArgumentFormatter.cls

jongpie committed:
Renamed QueryUtils, converted static methods to instance methods

I ended up using the name QueryArgumentFormatter - not sure that it's my absolute favorite, but I like it better that 'QueryUtils'

jongpie committed:
Merge branch 'dev' of https://github.com/jongpie/NebulaFramework into tech-debt/issue-20

jamessimone reviewed (approved):
Everything looks good!

jongpie committed:
Logger.cls improvements

Logger.cls is now static - the rest of the codebase has been updated to reflect this change

jongpie commented:
Gonna finally merge this guy with master soon. Here are the tentative release notes.

**Refactoring (Could Impact Existing Code)**
* Changed NebulaSettings.cls to use static variables and methods
* Changed Logger.cls to use static methods - everything now calls Logger directly
* Renamed QueryUtils.cls to QueryArgumentFormatter.cls (and refactored it)
* QueryBuilder.cls now uses QueryArgumentFormatter.cls
* Fixed the ordering of the filter scopes to be in descending order (for readability)

**Bugfixes**
* Fixed a problem with SObjectRecordTypes.cls when the object has no record types
* Logger now checks the custom setting to see if it is enabled

**Continuous Integration**
* Added .travis.yml for continuous integration. Passing unit tests with 75% code coverage is now required to approve a PR
* Added Travis CI build badges to README.md

**Unit Tests**
* Standardised the naming convention on a few test methods
* Added more unit tests & test classes
